scripts/generate_enums.py

```diff
@@ -0,0 +1,356 @@
+#! /usr/bin/env python3
+import argparse
+import os
+import string
+import textwrap
+import xml.etree.ElementTree as ElementTree
+
+CONFIG_FILE_PATH = "../build/docs/public_api/xml/config_8h.xml"
+
+ROC_JAVA_BASE_PATH = "../../roc-java"
+ROC_GO_BASE_PATH = "../../roc-go"
+
+ROC_JAVA_PACKAGE = "org.rocstreaming.roctoolkit"
+
+ODD_PREFIXES = {'roc_clock_source': 'ROC_CLOCK_', 'roc_protocol': 'ROC_PROTO_'}
+
+
+def to_camel_case(name):
+    return ''.join(x.capitalize() for x in name.split('_'))
+
+
+class DocComment:
+    def __init__(self, items: list):
+        self.items: list = items
+
+
+class EnumValue:
+    def __init__(self, name: string, value: string, doc: DocComment):
+        self.name: string = name
+        self.value: string = value
+        self.doc: DocComment = doc
+
+
+class EnumDefinition:
+    def __init__(self, name: string, values: list[EnumValue], doc: DocComment):
+        self.name: string = name
+        self.values: list[EnumValue] = values
+        self.doc: DocComment = doc
+
+
+class EnumGenerator:
+    def generate_enum(self, enum_definition: EnumDefinition):
+        pass
+
+
+class JavaEnumGenerator(EnumGenerator):
+
+    def __init__(self, base_path: string, name_prefixes: dict[str, str]):
+        self.base_path = base_path
+        self.name_prefixes: dict[str, str] = name_prefixes
+
+    def generate_enum(self, enum_definition: EnumDefinition):
+        java_name = self.get_java_enum_name(enum_definition.name)
+        enum_values = enum_definition.values
+
+        enum_file_path = (self.base_path + "/src/main/java/"
+                          + ROC_JAVA_PACKAGE.replace(".", "/") + "/" + java_name + ".java")
+        enum_file = open(enum_file_path, "w")
+
+        enum_file.write("// Code generated by generate_enums script from roc-toolkit; DO NOT EDIT!\n\n")
```

Comment:
I suggest to add roc-toolkit commit SHA and git tag, for which these enums were generated.

Reply:
done